### PR TITLE
Added some custom client-side form validation

### PR DESCRIPTION
**Pull request by kshade**
This adds client-side form validation for the nickname field using Javascript and browser functionality, as requested in https://github.com/eurofurence/crittersystem/issues/31

image

The upsides:

- Uses the same regex and error message as the server-side validation
- Compatible with accessibility tools (ARIA)
- Works (for my limited test cases)

Downsides:

- Not very pretty (but Bootstrap recommends not to use their implementation if you want accessibility)
- Might be overkill/dumb/insane.
- Just a static box with the validation message as information might be better.

### CODE REVIEW

File: resources/assets/js/forms.js
Comment (Corborax) [inline resources/assets/js/forms.js:1]: @Fenrikur Can you have a look at this regex, please?

**Review by Corborax — COMMENTED** [review]
Someone with more regex experience needs to look at it.

Thumbs up for keeping an eye on the AIRA. Accessibility is one of the goals that EF is aiming for.

File: resources/assets/js/forms.js
Comment (Corborax) [reply resources/assets/js/forms.js:1]: From config.php

     // Regular expression describing a FALSE username.
        // Per default usernames must only contain alphanumeric chars, "-", "_" or ".".
        'username_regex' => (string) env('USERNAME_REGEX', '/([^\p{L}\p{N}_.-]+)/ui'),
       
 
Maybe it makes sense to edit hat USERNAME_REGEX as environmentable variable?

If we allow a space, will it break the system? 
       
Why was it not allowed in the angel system?

Comment (RustyBraze) [reply resources/assets/js/forms.js:1]: > Maybe it makes sense to edit hat USERNAME_REGEX as environmentable variable?

Yes and No. Yes to allow the system to be flexible. No, should be inside the config file explicity (without checking the env).

> If we allow a space, will it break the system?

Yes and no. Yes for the login system and people tend to put more than 1 space between words. (This gives a chance for the user to create an issue and later we have to solve). No if you are using external authentication.

> Why was it not allowed in the angel system?

Usernames usually does not allow spaces. Common industry standard

**Review by RustyBraze — APPROVED** [review]
I agree with the solution implemented

File: resources/assets/js/forms.js
Comment (Corborax) [reply resources/assets/js/forms.js:1]: Thank you! @RustyBraze  :)

**Review by Corborax — APPROVED** [review]

**Pull request merged**